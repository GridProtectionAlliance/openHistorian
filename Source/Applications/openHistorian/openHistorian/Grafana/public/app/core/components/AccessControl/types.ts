export type ResourcePermission = {
  id: number;
  resourceId: string;
  isManaged: boolean;
  isInherited: boolean;
  userId?: number;
  userLogin?: string;
  userAvatarUrl?: string;
  team?: string;
  teamId?: number;
  teamAvatarUrl?: string;
  builtInRole?: string;
  actions: string[];
  permission: string;
};

export type SetPermission = {
  userId?: number;
  teamId?: number;
  builtInRole?: string;
  permission: string;
  target: PermissionTarget;
};

export enum PermissionTarget {
  None = 'None',
  Team = 'Team',
  User = 'User',
  BuiltInRole = 'builtInRole',
}
export type Description = {
  assignments: Assignments;
  permissions: string[];
};

export type Assignments = {
  users: boolean;
  teams: boolean;
  builtInRoles: boolean;
};
