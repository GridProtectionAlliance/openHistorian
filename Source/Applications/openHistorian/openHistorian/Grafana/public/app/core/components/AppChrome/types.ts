import { NavModelItem } from '@grafana/data';

export const TOP_BAR_LEVEL_HEIGHT = 40;

export interface ToolbarUpdateProps {
  pageNav?: NavModelItem;
  actions?: React.ReactNode;
}
