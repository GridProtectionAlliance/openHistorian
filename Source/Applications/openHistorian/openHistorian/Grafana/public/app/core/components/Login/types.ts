export interface LoginDTO {
  message: string;
  redirectUrl: string;
}
