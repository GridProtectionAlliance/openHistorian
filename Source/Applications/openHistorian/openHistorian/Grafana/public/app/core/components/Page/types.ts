import React, { FC, HTMLAttributes, RefCallback } from 'react';

import { NavModel, NavModelItem, PageLayoutType } from '@grafana/data';

import { OldNavOnly } from './OldNavOnly';
import { PageContents } from './PageContents';

export interface PageProps extends HTMLAttributes<HTMLDivElement> {
  children: React.ReactNode;
  navId?: string;
  navModel?: NavModel;
  pageNav?: NavModelItem;
  /** Can be used to place info inline with the heading */
  info?: PageInfoItem[];
  /** Can be used to place actions inline with the heading */
  actions?: React.ReactNode;
  /** Can be used to customize rendering of title */
  renderTitle?: (title: string) => React.ReactNode;
  /** Can be used to customize or customize and set a page sub title */
  subTitle?: React.ReactNode;
  /** Control the page layout. */
  layout?: PageLayoutType;
  /** Something we can remove when we remove the old nav. */
  toolbar?: React.ReactNode;
  /** Can be used to get the scroll container element to access scroll position */
  scrollRef?: RefCallback<HTMLDivElement>;
  /** Can be used to update the current scroll position */
  scrollTop?: number;
}

export interface PageInfoItem {
  label: string;
  value: React.ReactNode;
}

export interface PageType extends FC<PageProps> {
  OldNavOnly: typeof OldNavOnly;
  Contents: typeof PageContents;
}
