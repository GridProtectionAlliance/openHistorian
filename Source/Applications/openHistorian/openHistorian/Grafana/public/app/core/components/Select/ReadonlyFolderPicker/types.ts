import { PermissionLevelString } from '../../../../types';

export type PermissionLevel = Exclude<PermissionLevelString, PermissionLevelString.Admin>;
