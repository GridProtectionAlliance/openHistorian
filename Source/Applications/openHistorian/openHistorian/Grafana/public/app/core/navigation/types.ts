import React from 'react';
import { RouteComponentProps } from 'react-router-dom';

import { UrlQueryMap } from '@grafana/data';

export interface GrafanaRouteComponentProps<T extends {} = {}, Q = UrlQueryMap> extends RouteComponentProps<T> {
  route: RouteDescriptor;
  queryParams: Q;
}

export type GrafanaRouteComponent<T extends {} = any> = React.ComponentType<GrafanaRouteComponentProps<T>>;

export interface RouteDescriptor {
  path: string;
  component: GrafanaRouteComponent<any>;
  roles?: () => string[];
  pageClass?: string;
  /** Can be used like an id for the route if the same component is used by many routes */
  routeName?: string;
  chromeless?: boolean;
  exact?: boolean;
  sensitive?: boolean;
}
