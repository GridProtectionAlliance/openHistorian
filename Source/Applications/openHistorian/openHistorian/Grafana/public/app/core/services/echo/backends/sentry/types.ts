import { Event as SentryEvent } from '@sentry/browser';
import { Response } from '@sentry/types';

import { EchoEvent, EchoEventType } from '@grafana/runtime';

export interface BaseTransport {
  sendEvent(event: SentryEvent): PromiseLike<Response>;
}

export type SentryEchoEvent = EchoEvent<EchoEventType.Sentry, SentryEvent>;

export interface User {
  email: string;
  id: number;
  orgId: number;
}
