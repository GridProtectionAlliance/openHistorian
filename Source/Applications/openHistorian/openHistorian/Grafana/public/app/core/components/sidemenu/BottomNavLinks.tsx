import React, { PureComponent } from 'react';
import appEvents from '../../app_events';
import { User } from '../../services/context_srv';
import { NavModelItem } from '@grafana/data';
import { CoreEvents } from 'app/types';
import { OrgSwitcher } from '../OrgSwitcher';
import { getFooterLinks } from '../Footer/Footer';

export interface Props {
  link: NavModelItem;
  user: User;
}

interface State {
  showSwitcherModal: boolean;
}

class BottomNavLinks extends PureComponent<Props, State> {
  state: State = {
    showSwitcherModal: false,
  };

  onOpenShortcuts = () => {
    appEvents.emit(CoreEvents.showModal, {
      templateHtml: '<help-modal></help-modal>',
    });
  };

  toggleSwitcherModal = () => {
    this.setState(prevState => ({
      showSwitcherModal: !prevState.showSwitcherModal,
    }));
  };

  render() {
    const { link, user } = this.props;
    const { showSwitcherModal } = this.state;

    let children = link.children || [];

    if (link.id === 'help') {
      children = getFooterLinks();
    }

    return (
      <div className="sidemenu-item dropdown dropup">
        <a href={link.url} className="sidemenu-link" target={link.target}>
          <span className="icon-circle sidemenu-icon">
            {link.icon && <i className={link.icon} />}
            {link.img && <img src={link.img} />}
          </span>
        </a>
        <ul className="dropdown-menu dropdown-menu--sidemenu" role="menu">
          {link.subTitle && (
            <li className="sidemenu-subtitle">
              <span className="sidemenu-item-text">{link.subTitle}</span>
            </li>
          )}
          {link.showOrgSwitcher && (
            <li className="sidemenu-org-switcher">
              <a onClick={this.toggleSwitcherModal}>
                <div>
                  <div className="sidemenu-org-switcher__org-name">{user.orgName}</div>
                  <div className="sidemenu-org-switcher__org-current">Current Org:</div>
                </div>
                <div className="sidemenu-org-switcher__switch">
                  <i className="fa fa-fw fa-random" />
                  Switch
                </div>
              </a>
            </li>
          )}

          {showSwitcherModal && <OrgSwitcher onDismiss={this.toggleSwitcherModal} />}

          {children.map((child, index) => {
            return (
              <li key={`${child.text}-${index}`}>
                <a href={child.url} target={child.target} rel="noopener">
                  {child.icon && <i className={child.icon} />}
                  {child.text}
                </a>
              </li>
            );
          })}

          {link.id === 'help' && (
            <li key="keyboard-shortcuts">
              <a onClick={() => this.onOpenShortcuts()}>
                <i className="fa fa-keyboard-o" /> Keyboard shortcuts
              </a>
            </li>
          )}

          <li className="side-menu-header">
            <span className="sidemenu-item-text">{link.text}</span>
          </li>
        </ul>
      </div>
    );
  }
}

export default BottomNavLinks;
