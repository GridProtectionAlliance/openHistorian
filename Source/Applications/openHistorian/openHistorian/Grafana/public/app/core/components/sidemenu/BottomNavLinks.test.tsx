import React from 'react';
import { shallow } from 'enzyme';
import BottomNavLinks from './BottomNavLinks';
import appEvents from '../../app_events';
import { CoreEvents } from 'app/types';

jest.mock('../../app_events', () => ({
  emit: jest.fn(),
}));

const setup = (propOverrides?: object) => {
  const props = Object.assign(
    {
      link: {
        text: 'Hello',
      },
      user: {
        id: 1,
        isGrafanaAdmin: false,
        isSignedIn: false,
        orgCount: 2,
        orgRole: '',
        orgId: 1,
        login: 'hello',
        orgName: 'Grafana',
        timezone: 'UTC',
        helpFlags1: 1,
        lightTheme: false,
        hasEditPermissionInFolders: false,
      },
    },
    propOverrides
  );
  return shallow(<BottomNavLinks {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render organization switcher', () => {
    const wrapper = setup({
      link: {
        showOrgSwitcher: true,
      },
    });

    wrapper.find('.sidemenu-org-switcher a').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('should render subtitle', () => {
    const wrapper = setup({
      link: {
        subTitle: 'subtitle',
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render children', () => {
    const wrapper = setup({
      link: {
        children: [
          {
            id: '1',
          },
          {
            id: '2',
          },
          {
            id: '3',
          },
          {
            id: '4',
            hideFromMenu: true,
          },
        ],
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Functions', () => {
  describe('item clicked', () => {
    const wrapper = setup();
    it('should emit show modal event if url matches shortcut', () => {
      const instance = wrapper.instance() as BottomNavLinks;
      instance.onOpenShortcuts();

      expect(appEvents.emit).toHaveBeenCalledWith(CoreEvents.showModal, { templateHtml: '<help-modal></help-modal>' });
    });
  });
});
