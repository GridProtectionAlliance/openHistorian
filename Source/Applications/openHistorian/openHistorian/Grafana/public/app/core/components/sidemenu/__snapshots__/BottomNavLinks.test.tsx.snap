// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render children 1`] = `
<div
  className="sidemenu-item dropdown dropup"
>
  <a
    className="sidemenu-link"
  >
    <span
      className="icon-circle sidemenu-icon"
    />
  </a>
  <ul
    className="dropdown-menu dropdown-menu--sidemenu"
    role="menu"
  >
    <li
      key="undefined-0"
    >
      <a
        rel="noopener"
      />
    </li>
    <li
      key="undefined-1"
    >
      <a
        rel="noopener"
      />
    </li>
    <li
      key="undefined-2"
    >
      <a
        rel="noopener"
      />
    </li>
    <li
      key="undefined-3"
    >
      <a
        rel="noopener"
      />
    </li>
    <li
      className="side-menu-header"
    >
      <span
        className="sidemenu-item-text"
      />
    </li>
  </ul>
</div>
`;

exports[`Render should render component 1`] = `
<div
  className="sidemenu-item dropdown dropup"
>
  <a
    className="sidemenu-link"
  >
    <span
      className="icon-circle sidemenu-icon"
    />
  </a>
  <ul
    className="dropdown-menu dropdown-menu--sidemenu"
    role="menu"
  >
    <li
      className="side-menu-header"
    >
      <span
        className="sidemenu-item-text"
      >
        Hello
      </span>
    </li>
  </ul>
</div>
`;

exports[`Render should render organization switcher 1`] = `
<div
  className="sidemenu-item dropdown dropup"
>
  <a
    className="sidemenu-link"
  >
    <span
      className="icon-circle sidemenu-icon"
    />
  </a>
  <ul
    className="dropdown-menu dropdown-menu--sidemenu"
    role="menu"
  >
    <li
      className="sidemenu-org-switcher"
    >
      <a
        onClick={[Function]}
      >
        <div>
          <div
            className="sidemenu-org-switcher__org-name"
          >
            Grafana
          </div>
          <div
            className="sidemenu-org-switcher__org-current"
          >
            Current Org:
          </div>
        </div>
        <div
          className="sidemenu-org-switcher__switch"
        >
          <i
            className="fa fa-fw fa-random"
          />
          Switch
        </div>
      </a>
    </li>
    <OrgSwitcher
      onDismiss={[Function]}
    />
    <li
      className="side-menu-header"
    >
      <span
        className="sidemenu-item-text"
      />
    </li>
  </ul>
</div>
`;

exports[`Render should render subtitle 1`] = `
<div
  className="sidemenu-item dropdown dropup"
>
  <a
    className="sidemenu-link"
  >
    <span
      className="icon-circle sidemenu-icon"
    />
  </a>
  <ul
    className="dropdown-menu dropdown-menu--sidemenu"
    role="menu"
  >
    <li
      className="sidemenu-subtitle"
    >
      <span
        className="sidemenu-item-text"
      >
        subtitle
      </span>
    </li>
    <li
      className="side-menu-header"
    >
      <span
        className="sidemenu-item-text"
      />
    </li>
  </ul>
</div>
`;
